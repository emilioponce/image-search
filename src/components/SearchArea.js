import React from 'react'
import Search from '../containers/Search'
import ClearSearch from '../containers/ClearSearch'
import Status from '../containers/Status'

import './SearchArea.css'

const SearchArea = () => {
  return (
    <>
      <div className="SearchArea">
        <Search />
        <ClearSearch />
      </div>
      <Status />
    </>
  )
}

export default SearchArea
