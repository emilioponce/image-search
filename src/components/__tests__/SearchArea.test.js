import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import SearchArea from '../SearchArea'

jest.mock('../../containers/Search', () => 'mock-search')
jest.mock('../../containers/ClearSearch', () => 'mock-clearsearch')
jest.mock('../../containers/Status', () => 'mock-status')

describe('SearchArea component', () => {
  const component = <SearchArea />

  it('Renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
