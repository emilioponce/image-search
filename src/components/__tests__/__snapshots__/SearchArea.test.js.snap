// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchArea component Snapshot matchs 1`] = `
Array [
  <div
    className="SearchArea"
  >
    <mock-search />
    <mock-clearsearch />
  </div>,
  <mock-status />,
]
`;
